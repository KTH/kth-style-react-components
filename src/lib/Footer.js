import React from 'react'
import cortina from 'kth-node-cortina-block'

class Footer extends React.Component {
  constructor() {
    super()
    this.state = { footer: '' }
  }

  componentDidMount() {
    cortina({url: this.props.url})
      .then(blocks => this.setState({ footer: blocks.footer }))
  }

  render () {
    return <footer role="contentinfo" className="container">
      <div dangerouslySetInnerHTML={{__html: this.state.footer}} />
    </footer>
  }
}

Footer.defaultProps = {
  url: 'https://www-r.referens.sys.kth.se/cm/',
};

export default Footer