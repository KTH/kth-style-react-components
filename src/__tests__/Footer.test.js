import React from "react";
import renderer from "react-test-renderer";
import Footer from '../lib/Footer';

describe("Footer", () => {

  // TODO: await cortina api call
  it("should match previous snapshot", async () => {
    const tree = renderer
      .create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});