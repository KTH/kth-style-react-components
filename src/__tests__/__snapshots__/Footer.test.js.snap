// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match previous snapshot 1`] = `
<footer
  className="container"
  role="contentinfo"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</footer>
`;
